### ISSUE TITLE
Visdom Is Extremely Slow With Many Points

### ISSUE THREAD

**Issue by enchainingrealm** [facebookresearch/visdom#650]
I'm using Visdom to plot the training loss on every mini-batch onto a line plot. After only a few epochs (around 2000 mini-batches = 2000 points on the line plot,) my Visdom browser window becomes extremely slow and unresponsive.

Is there a way for Visdom to optimize plotting line plots containing many points?

**Comment by JackUrb** [comment]
Hi @enchainingrealm. As of now we don't have backend optimizers for averaging out points on dense plots to reduce density, and standard plotly plots choke under heavy load. Still, so long as you don't have too many of these dense plots, you can render using webgl as noted in our docs and demonstrated in our python demo.

**Comment by enchainingrealm** [comment]
Visdom is still slow with WebGL enabled, but now at least its usable. I'm closing this issue since you mentioned that averaging out points that live in the same x-coordinate (on screen) is not yet implemented, and that was what I was looking for.
